app/models/plate.rb

```diff
@@ -61,6 +61,14 @@ def construct!
     end
 
     # Returns the wells with their pool identifier included
+    # @note This is a method defined on the well association, and can be considered a scope.
+    # Ie. You can do plate.wells.with_pool_id.located_at(['A1'])
+    # It is a little special in that its behaviour is dependent on the {PlatePurpose#pool_wells plate purpose}.
+    # In practice this lets wells on PlatePurpose::Initial to pull their information from the requests, whereas
+    # other purposes jump back to the stock wells.
+    # Given this is usually called by the transfers themselves, prior to the transfer of the aliquots
+    # it can't be replaced purely by using request on aliquot (as it is partly responsible for setting that)
+    # however we can almost certainly simplify this whole process somewhat.
     def with_pool_id
```

Comment:
This was where my brain melted. I've tried to put a few pointers in the comments, and also added an issue  for a more long term fix.